Tests/TestFrameworks/NUnit4.Specs/FrameworkSpecs.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace NUnit4.Specs;
+
+[TestFixture]
+public class FrameworkSpecs
+{
+    [Test]
+    public void When_nunit4_is_used_it_should_throw_nunit_exceptions_for_assertion_failures()
+    {
+        // Act
+        Action act = () => 0.Should().Be(1);
+
+        // Assert
+        Exception exception = act.Should().Throw<Exception>().Which;
+        exception.GetType()
+            .FullName.Should()
+            .ContainEquivalentOf("NUnit.Framework.AssertionException");
```

Comment:
🤔 Are you sure this is the right one? I was also expecting a different version of the `NUnitTestFramework` type. Or are the exception types the same?

Reply:
Our current `NUnitTestFramework` also matches NUnit4, so no changes are needed on our side.

[NUnit.Framework.AssertionException](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Exceptions/AssertionException.cs)

![image](https://github.com/fluentassertions/fluentassertions/assets/919634/78758c7a-6cf3-4dea-87c9-9b6ac391cd2c)

Reply:
The Test adapter also remained unchanged, according to the [release notes](https://docs.nunit.org/articles/nunit/release-notes/framework.html#nunit-400---november-26-2023):
![image](https://github.com/fluentassertions/fluentassertions/assets/3438234/660c94da-122b-4234-a422-fc4da62a60a8)